### ISSUE TITLE
Env of multi intersections has a bug

### ISSUE THREAD

**Issue by AlvinLXS** [LucasAlegre/sumo-rl#21]
There is a bug in the env of multi intersections. For example,  in `experiments/ql_4x4grid.py`,
```python
env = SumoEnvironment(net_file='nets/4x4-Lucas/4x4.net.xml',
                          route_file='nets/4x4-Lucas/4x4c1c2c1c2.rou.xml',
                          use_gui=True,
                          num_seconds=80000,
                          max_depart_delay=0)
```
there are 16 intersections but the `single_agent` in `class SumoEnvironment(MultiAgentEnv):` is **Flase**
```python 
    def __init__(self, net_file, route_file, out_csv_name=None, use_gui=False, num_seconds=20000, max_depart_delay=100000, time_to_teleport=-1, delta_time=5, yellow_time=2, min_green=5, max_green=50, single_agent=False):
```
nothing has been changed, so the state returned by the reset function maybe wrong. 

Be careful to use multi intersections environments.

**Comment by LucasAlegre** [comment]
Hi,

The env is multiagent by default, ie., single_agent=False.
In this case, it is a multiagent environment (16 intersections) and therefore single_agent=False.

When single_agent, the reset and step method return only the obs of the single agent. When single_agent=False, the return is a dictionary of observations for each intersection.

**Comment by AlvinLXS** [comment]
Thanks for your timely reply and I have made some mistakes.

**Comment by AlvinLXS** [comment]
Hi,
Thanks for your timely reply and I was a little confused yesterday. I've been struggling with a problem for a week. After I trained an RL model, I tested it twice in this environment, but the final performance indicators were different and there were obvious differences between the two. May I ask how to solve this situation?



xiaoshuang__li@163.com
 
From: Lucas Alegre
Date: 2021-01-15 00:01
To: LucasAlegre/sumo-rl
CC: lixiaoshuang; Author
Subject: Re: [LucasAlegre/sumo-rl] Env of multi intersections has a bug (#21)
Hi,
The env is multiagent by default, ie., single_agent=False.
In this case, it is a multiagent environment (16 intersections) and therefore single_agent=False.
When single_agent, the reset and step method return only the obs of the single agent. When single_agent=False, the return is a dictionary of observations for each intersection.
